### PR TITLE
[Merged by Bors] - feat: Port Topology.UniformSpace.AbsoluteValue

### PR DESCRIPTION
**Pull request by casavaca**
Co-authored-by: Adam Topaz <github@adamtopaz.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/UniformSpace/AbsoluteValue.lean [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#align is_absolute_value.uniform_space IsAbsoluteValue.uniformSpace
+
+-- Porting note: new instance to help `failed to synthesize Nonempty { ε // ε > 0 }`
+/-- local instance helper -/
+local instance nonempty_gt_helper [LT α] [NoMaxOrder α] (a : α) : Nonempty { x // x > a } :=
+  nonempty_subtype.2 (exists_gt a)
+
+theorem mem_uniformity {s : Set (R × R)} :
+    s ∈ (uniformSpaceCore abv).uniformity ↔ ∃ ε > 0, ∀ {a b : R}, abv (b - a) < ε → (a, b) ∈ s := by
+  suffices (s ∈ ⨅ ε : { ε : 𝕜 // ε > 0 }, 𝓟 { p : R × R | abv (p.2 - p.1) < ε.val }) ↔ _
```
Comment (adamtopaz) [inline Mathlib/Topology/UniformSpace/AbsoluteValue.lean:88]:
```suggestion
theorem mem_uniformity {s : Set (R × R)} :
    s ∈ (uniformSpaceCore abv).uniformity ↔ ∃ ε > 0, ∀ {a b : R}, abv (b - a) < ε → (a, b) ∈ s := by
  suffices (s ∈ ⨅ ε : { ε : 𝕜 // 0 < ε }, 𝓟 { p : R × R | abv (p.2 - p.1) < ε.val }) ↔ _
```

I **think** this should fix the issue (and make the `helper` instance above unnecessary, hence fixing the lint error)

Comment (adamtopaz) [reply Mathlib/Topology/UniformSpace/AbsoluteValue.lean:88]: If you decide to go with this fix, then please also add a `porting note` explaining that the instance search didn't fire with `ε > 0`, but that it does fire with `0 < ε`.

Comment (casavaca) [reply Mathlib/Topology/UniformSpace/AbsoluteValue.lean]: Done. Thanks.

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**